raincloudy/controller.py

```diff
@@ -61,70 +63,41 @@ def __repr__(self):
         except AttributeError:
             return "<{0}: {1}>".format(self.__class__.__name__, self.id)
 
-    def post(self, ddata, url=SETUP_ENDPOINT, referer=SETUP_ENDPOINT):
-        """Method to update some attributes on namespace."""
-        headers = HEADERS.copy()
-        if referer is None:
-            headers.pop('Referer')
-        else:
-            headers['Referer'] = referer
-
-        # append csrftoken
-        if 'csrfmiddlewaretoken' not in ddata.keys():
-            ddata['csrfmiddlewaretoken'] = self._parent.csrftoken
-
-        req = self._parent.client.post(url, headers=headers, data=ddata)
-        if req.status_code == 200:
-            self.update()
-
-    def _get_cu_and_fu_status(self):
-        """Submit GET request to update information."""
-        # adjust headers
-        headers = HEADERS.copy()
-        headers['Accept'] = '*/*'
-        headers['X-Requested-With'] = 'XMLHttpRequest'
-        headers['X-CSRFToken'] = self._parent.csrftoken
-
-        args = '?controller_serial=' + self.serial \
-               + '&faucet_serial=' + self.faucet.serial
-
-        req = self._parent.client.get(STATUS_ENDPOINT + args,
-                                      headers=headers)
-
-        # token probably expired, then try again
-        if req.status_code == 403:
-            self._parent.login()
-            self.update()
-        elif req.status_code == 200:
-            self.attributes = req.json()
-        else:
-            req.raise_for_status()
-
     def _refresh_html_home(self):
         """
         Function to refresh the self._parent.html['home'] object
         which provides the status if zones are scheduled to
         start automatically (program_toggle).
         """
-        req = self._parent.client.get(HOME_ENDPOINT)
+        data = {
```

Comment:
Needs to dynamically populate everything in this dict so we don't start overriding user schedules. Page would 500 when I toggled the active controller if I didn't include everything :/

Reply:
Thinking further I probably could get this info from `setup` but I _think_ that requires two additional page loads to iterate through faucets. I really really want to minimize how hard we're hitting melnor here to be a good citizen of their resources. So kludging through the `home` toggling is probably the best option here.

Reply:
One last random thought: I only really need to load html on startup. We can purely poll the json endpoint after for status updates. I don't particularly care if downstream projects like HA necessarily update the name in real-time during an `update()`. Especially if doing that risks modifying user watering parameters